tensorflow_graphics/rendering/camera/perspective.py

```diff
@@ -262,34 +290,42 @@ def matrix_from_intrinsics(focal, principal_point, name=None):
     ValueError: If the shape of `focal`, or `principal_point` is not
     supported.
   """
-  with tf.compat.v1.name_scope(name, "perspective_matrix_from_intrinsics",
-                               [focal, principal_point]):
-    focal = tf.convert_to_tensor(value=focal)
-    principal_point = tf.convert_to_tensor(value=principal_point)
-
-    shape.check_static(
-        tensor=focal, tensor_name="focal", has_dim_equals=(-1, 2))
-    shape.check_static(
-        tensor=principal_point,
-        tensor_name="principal_point",
-        has_dim_equals=(-1, 2))
-    shape.compare_batch_dimensions(
-        tensors=(focal, principal_point),
-        tensor_names=("focal", "principal_point"),
-        last_axes=-2,
-        broadcast_compatible=False)
-
-    fx, fy = tf.unstack(focal, axis=-1)
-    cx, cy = tf.unstack(principal_point, axis=-1)
-    zero = tf.zeros_like(fx)
-    one = tf.ones_like(fx)
-    matrix = tf.stack((fx, zero, cx,
-                       zero, fy, cy,
-                       zero, zero, one),
-                      axis=-1)  # pyformat: disable
-    matrix_shape = tf.shape(input=matrix)
-    output_shape = tf.concat((matrix_shape[:-1], (3, 3)), axis=-1)
-    return tf.reshape(matrix, shape=output_shape)
+  with tf.compat.v1.name_scope(
+      name, "perspective_matrix_from_intrinsics", [focal, principal_point]
+  ):
+      focal = tf.convert_to_tensor(value=focal)
+      principal_point = tf.convert_to_tensor(value=principal_point)
+      skew_coefficient = tf.convert_to_tensor(value=skew_coefficient)
+
+      shape.check_static(tensor=focal, tensor_name="focal", has_dim_equals=(-1, 2))
+      shape.check_static(
+          tensor=principal_point,
+          tensor_name="principal_point",
+          has_dim_equals=(-1, 2),
+      )
+      shape.compare_batch_dimensions(
+          tensors=(focal, principal_point),
+          tensor_names=("focal", "principal_point"),
+          last_axes=-2,
+          broadcast_compatible=False,
+      )
+      skew_coefficient_shape = skew_coefficient.shape.as_list()
```

Comment:
Please use similar shape checking code to the principal_point and focal above, and add skew to the compare_batch_dimensions call. Alternatively, use the get_broadcasted_shape utility to broadcast each tensor to a common shape, as in for example https://github.com/tensorflow/graphics/blob/3dd63357f54f42ad60c6dc0daeb3e1739f7863be/tensorflow_graphics/geometry/transformation/rotation_matrix_3d.py#L398